hack/generator/pkg/codegen/swagger_type_extractor_test.go

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func Example_inferNameFromURLPath() {
+	group, name, _ := inferNameFromURLPath("/Microsoft.GroupName/resourceName/{resourceId}")
```

Comment:
Is this a test?

Sorta weird casing `_infer` vs `_Infer`.

Also... no assert?

Reply:
for an Example the specified magic Output comment is tested against what is what is written on stdout (and the Example goes in the docs, I think): see https://blog.golang.org/examples

Reply:
and re casing, if it doesn't match then a linter fires 😛

Reply:
TIL about Examples.